src/controller/CHIPDeviceController.cpp

```diff
@@ -760,6 +764,7 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
         }
     }
 
+    MATTER_LOG_METRIC_END(PASESession, err);
```

Comment:
Why not use RAII like `MATTER_TRACE_SCOPE` does? This can lead to error-prone code updates where MATTER_LOG_METRIC_END has to be called in every exit path properly.

Reply:
We already have a scope macro for this method and it already handles noop when tracing is disabled.

For cases when things are async, we can use new macros.